src/CodeExport/CodeExporter.class.st

```diff
@@ -12,9 +12,15 @@ Class {
 { #category : 'file-out' }
 CodeExporter class >> writeSourceCodeFrom: aStream baseName: baseName isSt: stOrCsFlag [
 
-	| extension targetFile |
-	extension := stOrCsFlag ifTrue: ['st']  ifFalse: ['cs'].
-	targetFile := FileLocator imageDirectory / baseName asFileName, extension.
+	| extension targetFile validFilename targetFilename |
+	
+	extension := stOrCsFlag ifTrue: ['.st'] ifFalse: ['.cs'].
+	
+	validFilename := FilePathEncoder encode: (FileSystem disk 
+		checkName: baseName
+		fixErrors: true).
+	targetFilename := (FilePathEncoder decode: validFilename) , extension.
```

Comment:
@hernanmd what I mean is to remove this here too!
There is no need to encode the paths, `FileSystem` and `FileLocator` do it already.

Reply:
Yes I did not have the time to look but this was my point.